scripts/lola-perf.ps1

```diff
@@ -0,0 +1,100 @@
+<#
+
+.SYNOPSIS
+This script runs performance tests for LoLa using secnetperf and generates results in a table.
+
+.PARAMETER SecNetPerfBinary
+    Specifies the secnetperf binary to use.
+
+.PARAMETER Target
+    Specifies -target parameter for secnetperf.
+
+.PARAMETER Bind
+    Specifies -bind parameter for secnetperf.
+
+.PARAMETER Responses
+    Specifies -response parameter for secnetperf.
+
+.PARAMETER NumIterations
+    Specifies the number of iterations to be run.
+#>
+
+param (
+    [Parameter(Mandatory = $true)]
+    [string]$SecNetPerfBinary,
+    [Parameter(Mandatory = $false)]
+    [string]$Target = "quic-server",
+    [Parameter(Mandatory = $false)]
+    [string]$Bind = "0.0.0.0",
+    [Parameter(Mandatory = $false)]
+    [Int32[]]$Responses = @(512, 1024, 4096, 8192, 16384, 32768, 65536),
+    [Parameter(Mandatory = $false)]
+    [Int32]$NumIterations = 3
+)
+
+class TestResult {
+    [string]$ResponseSize
+    [Int32]$Min
+    [Int32]$P50
+    [Int32]$P90
+    [Int32]$P99
+    [Int32]$P999
+    [Int32]$P9999
+}
+
+[Int32]$script:TotalNumTestCases = 0
+[Int32]$script:NumTestCasesCompleted = 0
+
+function RunTest (
+    [string]$ResponseSize,
+    [Int32]$NumIterations
+    )
+{
+    $Result = [TestResult]::new()
+    $Result.ResponseSize = $ResponseSize
+
+    [System.Collections.ArrayList]$Min = @()
+    [System.Collections.ArrayList]$P50 = @();
+    [System.Collections.ArrayList]$P90 = @();
+    [System.Collections.ArrayList]$P99 = @();
+    [System.Collections.ArrayList]$P999 = @();
+    [System.Collections.ArrayList]$P9999 = @();
+
+    for ($i = 0; $i -lt $NumIterations; $i++) {
+        $Output = Invoke-Expression  "$SecNetPerfBinary -test:rps -target:$Target -bind:$Bind -conns:1 -requests:1 -request:512 -response:$ResponseSize"
+        $MatchResults = $Output | Select-String -Pattern "Result: .*? RPS, Min: (.*?), Max: .*?, 50th: (.*?), 90th: (.*?), 99th: (.*?), 99.9th: (.*?), 99.99th: (.*?),"
+        if (!$MatchResults) {
+            Write-Error "Failed to parse secnetperf output"
+        }
+
+        $_ = $Min.Add([Int32]$MatchResults.Matches.Groups[1].Value)
```

Comment:
I'm confused... what do these "$_" mean? I've only seen these in like "$obj | where { $_.property -eq $something}"

Reply:
`$_` is the current item in the for-each loop

Reply:
You meant `$_` ? It's for capturing Add()'s return value. Otherwise, it prints the index of the element that was just added.

Reply:
It's like `_` in python.